src/Maestro/DependencyUpdater/DependencyUpdater.cs

```diff
@@ -217,32 +217,42 @@ private async Task CheckSubscriptionsAsync(UpdateFrequency targetUpdateFrequency
         {
             var enabledSubscriptionsWithTargetFrequency = (await Context.Subscriptions
                     .Where(s => s.Enabled)
-                    .Include(s => s.Channel)
-                    .ThenInclude(c => c.BuildChannels)
-                    .ThenInclude(bc => bc.Build)
-                    .ToListAsync())
-                .Where(s => s.PolicyObject.UpdateFrequency == targetUpdateFrequency);
+                    .ToListAsync(cancellationToken))
+                    .Where(s => s.PolicyObject.UpdateFrequency == targetUpdateFrequency);
 
             int subscriptionsUpdated = 0;
             foreach (var subscription in enabledSubscriptionsWithTargetFrequency)
             {
-                Build latestBuildInTargetChannel = subscription.Channel.BuildChannels.Select(bc => bc.Build)
+                Subscription subscriptionWithBuilds = await Context.Subscriptions
+                    .Where(s => s.Id == subscription.Id)
+                    .Include(s => s.Channel)
+                    .ThenInclude(c => c.BuildChannels)
```

Comment:
Is this loading every build that was ever applied to the channel?

Reply:
Yep. And we were previously doing it for all subscriptions at the same time which caused a giant pile of joins. I think I can probably restrict it to only get the last week or two of builds since our maximum frequency is a week between updates?

Reply:
I would choose something like 30 days? Unfortunately, it does mean that if a build hadn't been added to a channel in quite a while, triggering wouldn't work.

We could get clever and use a 2-phase lookup. You first look in the last 10 days or so, and then if no results are returned, you do the full query (or even something reasonable like 90 days)

Reply:
Thinking crazy, what about a database trigger that fires on any build info change and writes to a new table where we do these lookups.

Reply:
Possible, but more complicated, since builds can be removed from channels (meaning the newest build can change).